CONTRIBUTING.rst

```diff
@@ -0,0 +1,480 @@
+
+     $ git rebase -i <offending-commit-id>^
+
+   In the interactive rebase editor, replace ``pick`` with ``edit`` to select
+   a specific commit (if there's more than one in your pull request), or
+   remove the line to delete a commit entirely.  Then edit files to fix the
+   issues in the review.
+
+   As before, inspect and test your changes. When ready, continue the
+   patch submission::
+
+     $ git add [file(s)]
+     $ git rebase --continue
+
+   Update commit comment if needed, and continue::
+
+     $ git push --force origin fix_comment_typo
+
+   By force pushing your update, your original pull request will be updated
+   with your changes so you won't need to resubmit the pull request.
+
+
+Commit Guidelines
+*****************
+
+Changes are submitted as Git commits. Each commit message must contain:
+
+* A short and descriptive subject line that is less than 72 characters,
+  followed by a blank line. The subject line must include a prefix that
+  identifies the subsystem being changed, folowed by a colon, and a short
+  title, for example:  ``doc: update wiki references to new site``.
+  (If you're updating an existing file, you can use
+  ``git log <filename>`` to see what developers used as the prefix for
+  previous patches of this file.)
+
+* A change description with your logic or reasoning for the changes, followed
+  by a blank line.
+
+* A Signed-off-by line, ``Signed-off-by: <name> <email>`` typically added
+  automatically by using ``git commit -s``
+
+* If the change address a Jira issue, include a line of the form::
+
+      Jira: ZEP-xxx
+
+
+All changes and topics sent to GitHub must be well-formed, as described above.
+
+Commit Message Body
+===================
+
+When editing the commit message, please briefly explain what your change
+does and why it's needed. A change summary of ``"Fixes stuff"`` will be rejected. An
+empty change summary is only acceptable for trivial changes fully described by
+the commit title (e.g., ``doc: fix misspellings in CONTRIBUTING doc``)
+
+The description body of the commit message must include:
+
+* **what** the change does,
+* **why** you chose that approach,
+* **what** assumptions were made, and
+* **how** you know it works -- for example, which tests you ran.
+
+For examples of accepted commit messages, you can refer to the Zephyr GitHub
+`changelog <https://github.com/zephyrproject-rtos/zephyr/commits/master>`__.
+
+Other Commit Expectations
+=========================
+
+* Commits must build cleanly when applied on top of each other, thus avoiding
+  breaking bisectability.
+
+* Commits must pass the *scripts/checkpatch.pl* requirements.
+
+* Each commit must address a single identifiable issue and must be
+  logically self-contained. Unrelated changes should be submitted as
+  separate commits and pull requests.
+
+* When a change is included in the set to enable other changes, but it will
+  not be part of the final set, let the reviewers know this.
```

Comment:
still, i copied that from somewhere anyways, does not apply to how we do things.